src/commands/selectpatchoptionset.h

```diff
@@ -0,0 +1,40 @@
+/*---------------------------------------------------------------------------*\
+                          ____  _ _ __ _ __  ___ _ _
+                         |_ / || | '_ \ '_ \/ -_) '_|
+                         /__|\_, | .__/ .__/\___|_|
+                             |__/|_|  |_|
+\*---------------------------------------------------------------------------*/
+#ifndef ZYPPER_COMMANDS_SELECTPATCHOPTIONSET_INCLUDED
+#define ZYPPER_COMMANDS_SELECTPATCHOPTIONSET_INCLUDED
+
+#include "basecommand.h"
+#include "issue.h"
+#include "src/update.h"
+
+class SelectPatchOptionSet : public BaseCommandOptionSet
+{
+public:
+
+  enum AnyTypeMode
+  {
+    EnableAnyType, //< Enables the arguments to take optional args and select any issue type if no arg was given,
+                   //  also enables the generic --issue flag
+    DisableAnyType
+  };
+
+  SelectPatchOptionSet ( ZypperBaseCommand &parent, AnyTypeMode mode_r = DisableAnyType );
+
+ using BaseCommandOptionSet::BaseCommandOptionSet;
```

Comment:
Introduces `BaseCommandOptionSet::BaseCommandOptionSet(ZypperBaseCommand &parent )`.
Together with L25 `SelectPatchOptionSet ( ZypperBaseCommand &parent, AnyTypeMode mode_r = DisableAnyType )`:
```
In file included from /Local/ma/zypp/zypper/src/Command.cc:29:0:
/Local/ma/zypp/zypper/src/commands/patch.h:31:49: error: conversion from ‘<brace-enclosed initializer list>’ to ‘SelectPatchOptionSet’ is ambiguous
   SelectPatchOptionSet _selectPatchOpts { *this };
                                                 ^
/Local/ma/zypp/zypper/src/commands/patch.h:31:49: note: candidates are:
In file included from /Local/ma/zypp/zypper/src/commands/patch.h:12:0,
                 from /Local/ma/zypp/zypper/src/Command.cc:29:
/Local/ma/zypp/zypper/src/commands/selectpatchoptionset.h:27:30: note: SelectPatchOptionSet::SelectPatchOptionSet(ZypperBaseCommand&)
  using BaseCommandOptionSet::BaseCommandOptionSet;
                              ^
/Local/ma/zypp/zypper/src/commands/selectpatchoptionset.h:25:3: note: SelectPatchOptionSet::SelectPatchOptionSet(ZypperBaseCommand&, SelectPatchOptionSet::AnyTypeMode)
   SelectPatchOptionSet ( ZypperBaseCommand &parent, AnyTypeMode mode_r = DisableAnyType );
   ^
...
src/CMakeFiles/zypper_lib.dir/build.make:75: recipe for target 'src/CMakeFiles/zypper_lib.dir/Command.cc.o' failed
make[2]: *** [src/CMakeFiles/zypper_lib.dir/Command.cc.o] Error 1
```

Reply:
...and indent of the `using ...` line

Reply:
Dropped the using